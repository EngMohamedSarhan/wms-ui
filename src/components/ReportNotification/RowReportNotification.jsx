import { CheckCircleOutline, ErrorOutline } from "@mui/icons-material";
import { Divider, Grid, Typography } from "@mui/material";
import PropTypes from "prop-types";
import React from "react";

const RowReportNotification = ({ isSuccess, status, message }) => {
  return (
    <Grid container direction="column" mt={1}>
      <Grid container alignItems="center" gap={2}>
        <Typography
          fontWeight="500"
          color={isSuccess ? "success.main" : "error.main"}
        >
          {status}
        </Typography>
        {isSuccess ? (
          <CheckCircleOutline color="success" />
        ) : (
          <ErrorOutline color="error" />
        )}
      </Grid>
      <Typography mt={1} ml={1}>
        {message}
      </Typography>
      <Divider />
    </Grid>
  );
};

RowReportNotification.propTypes = {
  isSuccess: PropTypes.bool,
  status: PropTypes.string.isRequired,
  message: PropTypes.string,
};

export default RowReportNotification;
