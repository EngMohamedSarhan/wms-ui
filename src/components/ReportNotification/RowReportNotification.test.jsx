/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import React from "react";
import ShallowRenderer from "react-shallow-renderer";

import RowReportNotification from "./RowReportNotification";

describe("RowReportNotification Component", () =>
  test("RowReportNotification render", () => {
    const { getByText } = render(
      <RowReportNotification status="status" message="message" />
    );
    const status = getByText("status");
    const message = getByText("message");
    expect(status).toBeInTheDocument();
    expect(message).toBeInTheDocument();
  }));

describe("RowReportNotification Component", () =>
  test("RowReportNotification snapshot", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<RowReportNotification status="status" message="message" />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  }));
