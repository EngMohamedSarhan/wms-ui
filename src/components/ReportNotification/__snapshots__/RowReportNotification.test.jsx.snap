// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowReportNotification Component RowReportNotification snapshot 1`] = `
<ForwardRef(Grid)
  container={true}
  direction="column"
  mt={1}
>
  <ForwardRef(Grid)
    alignItems="center"
    container={true}
    gap={2}
  >
    <ForwardRef(Typography)
      color="error.main"
      fontWeight="500"
    >
      status
    </ForwardRef(Typography)>
    <Memo
      color="error"
    />
  </ForwardRef(Grid)>
  <ForwardRef(Typography)
    ml={1}
    mt={1}
  >
    message
  </ForwardRef(Typography)>
  <ForwardRef(Divider) />
</ForwardRef(Grid)>
`;

exports[`RowReportNotification snapshot RowReportNotification snapshot 1`] = `
<ForwardRef(Grid)
  container={true}
  direction="column"
  mt={1}
>
  <ForwardRef(Grid)
    alignItems="center"
    container={true}
    gap={2}
  >
    <ForwardRef(Typography)
      color="error.main"
      fontWeight="500"
    >
      status
    </ForwardRef(Typography)>
    <Memo
      color="error"
    />
  </ForwardRef(Grid)>
  <ForwardRef(Typography)
    ml={1}
    mt={1}
  >
    message
  </ForwardRef(Typography)>
  <ForwardRef(Divider) />
</ForwardRef(Grid)>
`;
